### PR TITLE
Add IRSensor

### PR DESCRIPTION
**Pull request by bryanlai00**

### CODE REVIEW

File: src/main/java/frc/util/IRSensor.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    private static DigitalInput cubeSensor;
+
+    // Create instance of a timer that we can use to keep track of how long the
+    // gear is kept in the position for.
+//    private Timer timeSinceEntry;
+    // We manually record whether the timer is running (Timer keeps that information private...)
+//    private boolean isTimerRunning;
+
+    public IRSensor(int port) {
+        cubeSensor = new DigitalInput(port);
```
Comment (justinkim) [inline src/main/java/frc/util/IRSensor.java]: Cube sensor?

Comment (huiminwu) [reply src/main/java/frc/util/IRSensor.java]: :joy:

File: src/main/java/frc/util/IRSensor.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    //Power Regression used
+    /******************************************************************************
+     * IRSensor Constants
+     *****************************************************************************/
+    /******************************************************************************
+     * Note that these constants were copied from the RobotMap.java from Wildcard.
+     * These require further testing.
+     *****************************************************************************/
+
+    //TODO: Test the IR Sensor Threshold value. Should correspond with how far away the cube is from the IR sensor.
```
Comment (justinkim) [inline src/main/java/frc/util/IRSensor.java]: What cube?

File: src/main/java/frc/util/IRSensor.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    //Use SmartDashboard, track the IR Sensor Voltage value.
+    //The current value is accurate for when the cube is about 4 inches away from the IR sensor.
```
Comment (justinkim) [inline src/main/java/frc/util/IRSensor.java]: 🤔

File: src/main/java/frc/util/IRSensor.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+//    public static final double IR_SENSOR_THRESHOLD = 1;
+//    public static final double IR_TIME_IN_MECHANISM_THRESHOLD = 1.0;
+//
+//    public static final double EQUATION_FACTOR = 12.23368994;
+//    public static final double EQUATION_EXPONENT = -0.9779601588;
+//    public static final double CONVERSION_FACTOR_CM_TO_INCHES = 0.393701;
+
+    private static DigitalInput catchSensor;
```
Comment (justinkim) [inline src/main/java/frc/util/IRSensor.java]: Idk what a catch is, but in this game, there are Panels and Cargo.

**Comment by justinkim** [comment]
@bryanlai00  Would recommend a squash of your commits.

File: src/main/java/frc/util/IRSensor.java (hunk 4 above)
Comment (bryanlai00) [reply src/main/java/frc/util/IRSensor.java]: It's a combination of Cargo and Hatch ;)

Comment (bryanlai00) [reply src/main/java/frc/util/IRSensor.java]: Unless you want: cargoAndPanelSensor?

Comment (justinkim) [reply src/main/java/frc/util/IRSensor.java]: carnel or patch, please!

**Pull request merged**